File: lib/java/src/main/java/io/github/brenbar/uapi/internal/_MockInvocation.java
package io.github.brenbar.uapi.internal;

import java.util.Map;

public class _MockInvocation {
    public final String functionName;
    public final Map<String, Object> functionArgument;
    public boolean verified = false;

    public _MockInvocation(String functionName, Map<String, Object> functionArgument) {
        this.functionName = functionName;
        this.functionArgument = functionArgument;
    }
}

File: lib/ts/uapi/internal/mock/MockInvocation.ts
export class MockInvocation {
    functionName: string;
    functionArgument: Record<string, any>;
    verified: boolean;

    constructor(functionName: string, functionArgument: Record<string, any>) {
        this.functionName = functionName;
        this.functionArgument = functionArgument;
        this.verified = false;
    }
}
